src/scope.rs

```diff
@@ -1797,10 +1800,13 @@ pub(crate) mod data {
   }
 
   #[derive(Debug, Clone, Copy, Eq, PartialEq)]
+  #[repr(u8)]
   enum ScopeStatus {
     Free,
-    Current { zombie: bool },
-    Shadowed { zombie: bool },
+    CurrentZombie,
```

Comment:
If we `#[cfg(debug_assertions)]` here, the compiler may have more opportunities to optimize out the zombie code in release mode.